exatomic/va/va.py

```diff
@@ -0,0 +1,327 @@
+        to have a global average displacement of 0.04 Bohr.
+
+        When delta_type = 2 we normalize each displacement so every atom has a maximum
+        displacement of 0.04 Bohr on every normal mode.
+
+        Args:
+            freq (:class:`exatomic.atom.Frequency`): Frequency dataframe
+            delta_type (int): Integer value to define the type of delta parameter to use
+        """
+        # average displacement of 0.04 bohr for each normal mode
+        nat = len(freq['label'].drop_duplicates())
+        nmode = len(freq['freqdx'].drop_duplicates())
+        freqdx = freq['freqdx'].values
+        if delta_type == 0:
+            # Code using for loops
+            # a = np.linalg.norm(freq[['dx', 'dy', 'dz']].
+            #                                    values, axis=1, ord=2)
+            # vec_sum = []
+            # for i in range(0,len(a),3):
+            #     vec_sum.append([0])
+            #     for j in range(i,i+3):
+            #         vec_sum[-1] += a[j]
+            # print(a)
+            # print(vec_sum)
+            d = freq.groupby(['freqdx', 'frame']).apply(
+                lambda x: np.sum(np.linalg.norm(
+                    x[['dx', 'dy', 'dz']].values, axis=1))).values
+            delta = 0.04 * nat / d
+            delta = np.repeat(delta, nat)
+    
+        # global avrage displacement of 0.04 bohr for all atom displacements
+        elif delta_type == 1:
+            d = np.sum(np.linalg.norm(
+                freq[['dx', 'dy', 'dz']].values, axis=1))
+            delta = 0.04 * nat * nmode / (np.sqrt(3) * d)
+            delta = np.repeat(delta, nat*nmode)
+    
+        # maximum displacement of 0.04 bohr for any atom in each normal mode
+        elif delta_type == 2:
+            d = freq.groupby(['freqdx', 'frame']).apply(lambda x:
+                np.amax(abs(np.linalg.norm(x[['dx', 'dy', 'dz']].values, axis=1)))).values
+            delta = 0.04 / d
+            delta = np.repeat(delta, nat)
+        self.delta = pd.DataFrame.from_dict({'delta': delta, 'freqdx': freqdx})
+        #self.delta['delta'] *= Length['Angstrom', 'au']
+
+    def _gen_displaced(self, freq, atom, fdx):
+        # get needed data from dataframes
+        eqcoord = atom[['x', 'y', 'z']].values
+        symbols = atom['symbol'].values
+        znums = atom['Zeff'].values
+        if fdx == -1:
+            freq_g = freq.copy()
+        else:
+            freq_g = freq.groupby('freqdx').filter(lambda x: fdx in
+                                                    x['freqdx'].drop_duplicates().values).copy()
+        disp = freq_g[['dx','dy','dz']].values
+        modes = freq_g['frequency'].drop_duplicates().values
+        nat = len(eqcoord)
+        freqdx = freq_g['freqdx'].drop_duplicates().values
+        tnmodes = len(freq['freqdx'].drop_duplicates())
+        nmodes = len(freqdx)
+        # chop all values less than 1e-6
+        eqcoord[abs(eqcoord) < 1e-6] = 0.0
```

Comment:
It could be worth making the tolerance a class attribute (e.g. `self._tol` or `self.tol`). Gives the "power" user more flexibility

Reply:
Yeah, I forgot to add it in here. Probably since I don't use it as much as in the gaussian.output.

Reply:
I added it in for the next push.